### PR TITLE
chore(docs): setup docsite previews [INFENG-184]

### PR DESCRIPTION
**Pull request by loksonarius**
## Description

This PR sets up both:
- local tooling to publish docsite previews based off a current build
- automatic docsite generation for PRs that make docs changes

## Test Plan

Local scripts tested manually during development.
CircleCI changes tested by making a PR.
GHA changes require merging before being able to iterate on it. I'd do this locally, but given it depends on having a PR opened in the first place to interact with...

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

INFENG-184

### CODE REVIEW

File: docs/deploy/promote.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+import pathlib
+import time
+
+import boto3
+
+HERE = pathlib.Path(__file__).parent
+
+EXCLUDES = ["release-notes/", "attributions.xml"]
+
+BUILD = str(HERE / ".." / "site" / "xml")
```
Comment (dannysauer) [inline docs/deploy/promote.py:12]: No os.path.join here? So disappointing. ;)

File: docs/deploy/upload.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    parser.add_argument(
+        "--preview",
+        action="store_true",
+        default=True,
+        help="whether the upload should go under the short-lived /previews path",
+    )
+    parser.add_argument(
+        "--version-file",
+        type=str,
+        default=HERE / ".." / ".." / "VERSION",
```
Comment (dannysauer) [inline docs/deploy/upload.py:29]: Does this syntax actually work? I've never seen it, and Python3 on my phone complains about dividing strings by each other.

**Review by dannysauer — APPROVED** [review]
Assuming that string division format is ok, this all looks pretty great. I like that removal is initially disabled, and the extra safety checks which are in there. Great work!

**Comment by dannysauer** [comment]
It's a minor thing, but I suppose for completeness I'll call out the use of string.format. There's at least one place where using an f-string would probably be preferable as you've got several parameters; it's mildly annoying to keep the arguments in order if another parameter is added later, and I think the general consensus is that f-strings are easier to read.  Most if not all of the other python in the project uses f-strings, so that's probably good for consistency.

I doubt anyone is confused by the use of format, though. So unless someone else objects, that's more of a suggestion for the next iteration than a request for immediate change.

File: docs/deploy/upload.py (hunk 2 above)
Comment (loksonarius) [reply docs/deploy/upload.py:29]: `HERE` is a Path, not a string

Comment (dannysauer) [reply docs/deploy/upload.py:29]: Ah, that makes more sense. 😂

**Pull request merged**